web/docs/advanced/links.md

```diff
@@ -69,18 +107,36 @@ const linkToTask = routes.TaskRoute.build({ params: { id: 1 } })
 
 This will result in a link like this: `/task/1`.
 
-You can also pass `search` and `hash` props to the `build` function. Check out the [API Reference](#routes-object) for more details.
+### Optional Static Segments
+
+If a route contains optional static segments, you'll need to specify one of the possible paths:
+
+```wasp title="main.wasp"
+route OptionalRoute { path: "/task/:id/details?", to: OptionalPage }
+page OptionalPage { ... }
+```
 
+```tsx title="TaskList.tsx"
+const linkToOptional = routes.OptionalRoute.build({
+  path: '/task/:id/details', // or '/task/:id'
+  params: { id: 1 },
+})
+```
+
+You can also pass `search` and `hash` props to the `build` function. Check out the [API Reference](#routes-object) for more details.
 
 ## API Reference
 
 ### `Link` Component
 
 The `Link` component accepts the following props:
+
 - `to` <Required />
 
   - A valid Wasp Route path from your `main.wasp` file.
 
+    In the case of optional static segments, you must provide one of the possible paths which include or exclude the optional segment. For example, if the path is `/task/:id/details?`, you must provide either `/task/:id/details` or `/task/:id`.
```

Comment:
Questionmarks at the end missing I think.

Reply:
Also, what about this:
```typescript
type Test = ExpandRouteOnOptionalStaticSegments<'/a?'> 
```
Should it be `'/a'` or `'/' | '/a'`?

Reply:
> Questionmarks at the end missing I think.

What do you mean exactly? 

> type Test = ExpandRouteOnOptionalStaticSegments<'/a?'> 

I believe this is an edge case I didn't handle, I'll explore the React Router behaviour and adjust the code.

Reply:
Questionmarks -> same story as here: https://github.com/wasp-lang/wasp/pull/2256#discussion_r1773519169

Reply:
Aw yes, this is written as intended. `/task/:id/details?` expands into `/task/:id/details` or `/task/:id`.